### ISSUE TITLE
`consistent-boolean-name`: add `skipPrefixes` for framework required prefixes

### ISSUE THREAD

**Issue by eranhirsch** [sindresorhus/eslint-plugin-unicorn#3486]
### Description

The rule misfires on react hooks that return a boolean because they have to begin with `use` (e.g. `**use**IsMyBooleanHook`). The rule is still useful for React hooks so the `ignore` option is overkill; instead, we just want to ignore the prefix itself, and allow the rule to consider the rest of the name for validity.

```ts
  "unicorn/consistent-boolean-name": ["error", { skipPrefixes: ["use"] }],
```

### Examples

```ts
// ❌
declare function useMyFlag(): boolean;
declare function useIsCool(): number;

// ✅
declare function useIsMyFlag(): boolean;
declare function useCool(): number;
```

### Additional Info

There is currently no workaround except *redefining* the list of acceptable prefixes manually and add `use` to each one in a loop inside the config, this isn't portable or resilient to future changes to the current list of prefixes.
